sql-plugin/src/main/scala/com/nvidia/spark/rapids/RapidsBufferCatalog.scala

```diff
@@ -598,6 +612,18 @@ class RapidsBufferCatalog(
       stream: Cuda.Stream): Unit = {
     val spillStoreMaxSize = spillStore.getMaxSize
     if (spillStoreMaxSize.isDefined) {
+      // Now only host memory store has a max limitation, and will spill its buffers to disk,
+      // Only spill to disk if allowed.
+      println(s"->allow task ${TaskContext.get.taskAttemptId()} spilling " +
+        s"to disk? $allowSpillingToDisk")
+      if (!allowSpillingToDisk) {
+        if (spillStore.currentSize + buffer.getMemoryUsedBytes > spillStoreMaxSize.get) {
+          throw new OutOfMemoryError(s"HostMemoryStore OOM, max size ${spillStoreMaxSize.get}")
```

Comment:
nit: For this experiment it might be nice to throw an exception other than an OutOfMemoryError. Just to distinguish between this and running out of host memory.  This is really minor. This is mostly because on line 534 I got confused thinking who is going to throw an OOM?